iidm/iidm-api/src/main/java/com/powsybl/iidm/network/identifiers/IdWithWildcardsNetworkElementIdentifier.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.identifiers;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.Network;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.powsybl.iidm.network.identifiers.NetworkElementIdentifier.IdentifierType.ID_WITH_WILDCARDS;
+
+/**
+ * @author Etienne Lesot {@literal <etienne.lesot at rte-france.com>}
+ *
+ * identifier that allows to get a identifiable that have some unknown character
+ * for example when a number changer in an id of an Identifiable
+ *
+ */
```

Comment:
The javadoc should contain more information such as the wildcard characters, the allowed other characters, …

Plus, it would be nice to follow the common usage: with capital letters and punctuation, the `@author` tag placed after the description…